### PR TITLE
Add configurations for ALB rule creating custom resource

### PR DESCRIPTION
**Pull request by tthyer**
Adds develop and prod sceptre configurations for new custom resource that adds listener rules for an ALB

Depends on https://github.com/Sage-Bionetworks-IT/cfn-cr-alb-rule/pull/1 https://github.com/Sage-Bionetworks/scipool-infra/pull/148

### CODE REVIEW

File: config/develop/cfn-cr-alb-rule.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  Department: "Platform"
+  Project: "Infrastructure"
+  OwnerEmail: "it@sagebase.org"
+parameters:
+  OidcClientSecretKeyName: '/alb-notebook-access/AuthenticateOidcClientSecret'
+  OidcIssuer: 'https://repo-prod.prod.sagebase.org/auth/v1'
+  OidcAuthorizationEndpoint: 'https://signin.synapse.org'
+  OidcTokenEndpoint: 'https://qtg2zn2bbf.execute-api.us-east-1.amazonaws.com/token'
+  OidcUserInfoEndpoint: 'https://repo-prod.prod.sagebase.org/auth/v1/oauth2/userinfo'
+  OidcClientId: '100050'
```
Comment (tthyer) [inline config/develop/cfn-cr-alb-rule.yaml:13]: @ahayden and @brucehoff , please verify that these are the final values for these parameters, in this develop configuration and the prod configuration below.

Comment (brucehoff) [reply config/develop/cfn-cr-alb-rule.yaml:13]: OidcTokenEndpoint: 'https://repo-prod.prod.sagebase.org/auth/v1/oauth2/token'

(FYI:  All the endpoints can be found in the OIDC 'discovery document' at <issuer>/.well-known/openid-configuration, for Synapse, it's https://repo-prod.prod.sagebase.org/auth/v1/.well-known/openid-configuration .)

Comment (brucehoff) [reply config/develop/cfn-cr-alb-rule.yaml:13]: The 'dev' OAuth client ID is 100055, please make sure you also associate the client secret for this client (found in last pass) with '/alb-notebook-access/AuthenticateOidcClientSecret'

File: config/prod/cfn-cr-alb-rule.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  Department: "Platform"
+  Project: "Infrastructure"
+  OwnerEmail: "it@sagebase.org"
+parameters:
+  OidcClientSecretKeyName: '/alb-notebook-access/AuthenticateOidcClientSecret'
+  OidcIssuer: 'https://repo-prod.prod.sagebase.org/auth/v1'
+  OidcAuthorizationEndpoint: 'https://signin.synapse.org'
+  OidcTokenEndpoint: 'https://qtg2zn2bbf.execute-api.us-east-1.amazonaws.com/token'
+  OidcUserInfoEndpoint: 'https://repo-prod.prod.sagebase.org/auth/v1/oauth2/userinfo'
+  OidcClientId: '100055'
```
Comment (brucehoff) [inline config/prod/cfn-cr-alb-rule.yaml:13]: OidcTokenEndpoint: 'https://repo-prod.prod.sagebase.org/auth/v1/oauth2/token'

(FYI: All the endpoints can be found in the OIDC 'discovery document' at /.well-known/openid-configuration, for Synapse, it's https://repo-prod.prod.sagebase.org/auth/v1/.well-known/openid-configuration .)


The 'prod' OIDC client is 100053.  Pleasevmake sure you also associate the client secret for this client (found in last pass) with '/alb-notebook-access/AuthenticateOidcClientSecret'

File: config/develop/cfn-cr-alb-rule.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+template_path: remote/cfn-cr-alb-rule.yaml
+stack_name: cfn-cr-alb-rule
+stack_tags:
+  Department: "Platform"
+  Project: "Infrastructure"
+  OwnerEmail: "it@sagebase.org"
+parameters:
+  OidcClientSecretKeyName: '/alb-notebook-access/AuthenticateOidcClientSecret'
```
Comment (zaro0508) [inline config/develop/cfn-cr-alb-rule.yaml:8]: I believe this is different for develop and prod.  Also we already have this secret in the SSM, it's `/synapse-login-scipooldev/SynapseOauthClientSecret`.  you can find the value for 100053 (prod) and 100055 (develop) in lastpass under `Oauth Client for Service Catalog` in the Shared-Service-Catalog folder.

File: config/develop/cfn-cr-alb-rule.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  OidcIssuer: 'https://repo-prod.prod.sagebase.org/auth/v1'
```
Comment (zaro0508) [inline config/develop/cfn-cr-alb-rule.yaml:9]: hmm, there's already a duplicate of `OidcIssuer` and `OidcClientId` at https://github.com/Sage-Bionetworks/scipool-infra/blob/develop/config/develop/synapse-oidc-idp.yaml#L8  

Hmm, wondering if we should do a reference here?  or maybe that should be left as a refactor for another time?

File: config/develop/cfn-cr-alb-rule.yaml (hunk 1 above)
Comment (tthyer) [reply config/develop/cfn-cr-alb-rule.yaml:13]: The client IDs have been corrected. I will have to re-add the Dev secret and add the Prod secret after the KMS key being used is created when this stack is merged.

Comment (brucehoff) [reply config/develop/cfn-cr-alb-rule.yaml:13]: "The client IDs have been corrected."  I don't see a new commit with the changes...

Comment (tthyer) [reply config/develop/cfn-cr-alb-rule.yaml]: forgot to push, corrected

**Comment by support-dpulls** [comment]
Hey there, Walid from Dpulls here!
If you are wondering why Dpulls didn't create the status check for this PR, this is because the dependency PR referenced in the PR's body belong to a different organisation, which is not currently supported. We are tracking this issue here https://github.com/dpulls/backlog/issues/5 and it should be fixed in the upcoming days.
We use this public [backlog](https://github.com/dpulls/backlog)  to track any issue or get any feedback from our users. So if you have any problem, just open an issue and we will be more than willing to help!
Thanks!
cc @zaro0508

**Pull request merged**